src/sentry/tasks/update_user_reports.py

```diff
@@ -19,16 +19,29 @@ def update_user_reports(**kwargs):
         group__isnull=True, environment__isnull=True, date_added__gte=now - timedelta(days=1)
     )
 
-    count = 0
-    updated = 0
-    for report in user_reports:
-        count += 1
-        event = eventstore.get_event_by_id(report.project_id, report.event_id)
-        if event:
+    project_ids = [r.project_id for r in user_reports]
+    event_ids = [r.event_id for r in user_reports]
+    report_by_event = {r.event_id: r for r in user_reports}
+
+    snuba_filter = eventstore.Filter(project_ids=project_ids, event_ids=event_ids)
```

Comment:
You might want to do 1 query per project. The primary key for events only ensures that event_id is unique *within* a project. While it *should* be globally unique there is a chance it won't be and you're update will do the wrong thing.

Reply:
Gotcha. OK will do.